front/src/hooks/useBoxesViewFilters.ts

```diff
@@ -0,0 +1,159 @@
+  gender_id?: string;
+  box_state?: string[];
+  tag_ids?: string[];
+}
+
+// Create atoms for individual search parameters
+const locationIdAtom = atomWithSearchParams("location_id", "");
+const categoryIdAtom = atomWithSearchParams("category_id", "");
+const productIdAtom = atomWithSearchParams("product_id", "");
+const sizeIdAtom = atomWithSearchParams("size_id", "");
+const genderIdAtom = atomWithSearchParams("gender_id", "");
+const boxStateAtom = atomWithSearchParams("box_state", "");
+const tagIdsAtom = atomWithSearchParams("tag_ids", "");
+
+export const useBoxesViewFilters = () => {
+  const [locationId, setLocationId] = useAtom(locationIdAtom);
+  const [categoryId, setCategoryId] = useAtom(categoryIdAtom);
+  const [productId, setProductId] = useAtom(productIdAtom);
+  const [sizeId, setSizeId] = useAtom(sizeIdAtom);
+  const [genderId, setGenderId] = useAtom(genderIdAtom);
+  const [boxState, setBoxState] = useAtom(boxStateAtom);
+  const [tagIds, setTagIds] = useAtom(tagIdsAtom);
+
+  // Convert URL filters to react-table Filters format
+  const tableFilters = useMemo((): Filters<any> => {
+    const result: Filters<any> = [];
+
+    if (locationId) {
+      result.push({ id: "location", value: [locationId] });
+    }
+
+    if (categoryId) {
+      result.push({ id: "productCategory", value: [categoryId] });
+    }
+
+    if (productId) {
+      result.push({ id: "product", value: [productId] });
+    }
+
+    if (sizeId) {
+      result.push({ id: "size", value: [sizeId] });
+    }
+
+    if (genderId) {
+      result.push({ id: "gender", value: [genderId] });
+    }
+
+    if (boxState) {
+      const states = boxState.split(",").filter(Boolean);
+      if (states.length > 0) {
+        result.push({ id: "state", value: states });
+      }
+    }
+
+    if (tagIds) {
+      const tags = tagIds.split(",").filter(Boolean);
+      if (tags.length > 0) {
+        result.push({ id: "tags", value: tags });
+      }
+    }
+
+    return result;
+  }, [locationId, categoryId, productId, sizeId, genderId, boxState, tagIds]);
+
+  // Update a specific filter
+  const updateFilter = (key: keyof BoxesViewFilters, value: string | string[] | undefined) => {
+    switch (key) {
+      case "location_id":
+        setLocationId((value as string) || "");
+        break;
+      case "category_id":
+        setCategoryId((value as string) || "");
+        break;
+      case "product_id":
+        setProductId((value as string) || "");
+        break;
+      case "size_id":
+        setSizeId((value as string) || "");
+        break;
+      case "gender_id":
+        setGenderId((value as string) || "");
+        break;
+      case "box_state":
+        setBoxState(Array.isArray(value) ? value.join(",") : (value as string) || "");
+        break;
+      case "tag_ids":
+        setTagIds(Array.isArray(value) ? value.join(",") : (value as string) || "");
+        break;
+    }
+  };
+
+  // Clear all filters
+  const clearFilters = () => {
```

Comment:
I don't see any usage of the clearFilters and clearFilter functions, I think they can be removed.